### DISCUSSION TITLE
Compression of images

### DISCUSSION THREAD

**Discussion by flamuren** [dompdf/dompdf#3050] [General]
Hi,

I am a total beginner so please bare with me.

I am building my first website using wordpress and its a job board. I want a feature that lets the user download their resume and it will look good and be in PDF. I have got help setting this up as a plugin using dompdf system.

The only issue is that we have had problem with the images. So the images (and all info) is pulled from the online-resumes when pressing download. When we first tested their built plugin the images looked super strange/pixeled/buggy, but the coders fixed this somehow. The problem now is instead that the file size is big (around 4mb) which is too big for a resume, since some systems wont allow resumes over a certain size. So I asked the developers to check if its possible to compress the images using dompdf or the GD image processing you recommend and make it somewhere in the middle.

Below is a example resume using both profile picture and cover image and that makes the file very big. Is it possible to get it less big but still look good?

[oskar-hanborg-cv-1667380329.pdf](https://github.com/dompdf/dompdf/files/9920105/oskar-hanborg-cv-1667380329.pdf)

Thanks for a awesome plugin/system. I hope it will help a lot of people looking for jobs in Sweden 🙏

/Flamuren

**Comment by bsweeney** [comment]
Dompdf itself doesn't perform image optimization. There is some logic that redraws images based on placement ([background images](https://github.com/dompdf/dompdf/blob/v2.0.1/src/Renderer/AbstractRenderer.php#L281)) and type ([PNG](https://github.com/dompdf/dompdf/blob/v2.0.1/lib/Cpdf.php#L5810)) but generally speaking no logic specifically related to image optimization.

Your best bet would be to optimize your images prior rendering the PDF, but you could try utilizing knowledge of how Dompdf processes images to try and achieve better optimization. For example, compare the PDF size after rendering the following two documents:

```
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
</head>
<body>
  <img src="https://filesamples.com/samples/image/jpg/sample_5184%C3%973456.jpg" style="width: 100%;">
</body>
</html>
```

```
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
</head>
<body>
  <div style="background-image: url(https://filesamples.com/samples/image/jpg/sample_5184%C3%973456.jpg); width: 100%; height: 100%; background-size: 100% auto; background-repeat: no-repeat;">&nbsp;</div>
</body>
</html>
```

Things are maybe optimized a bit too much for that image when used as a background, but you can improve the resolution by increasing the Dompdf DPI. Increasing the DPI may require tweaking other styling, though, depending on how you achieve your layout since all non-real units (such as px) will be translated.

**Reply by flamuren** [reply]
Many thanks for your answer 🙏

Aha thanks for clearing that out. I thought dompdf or GD extension could compress images to keep the file size down. This is all new to me so I am just trying to understand how this works.

The coders tried one without DPI (500kB filesize) where the images looks a bit strange and one with DPI (3mb) where the images looks good. My head cant grasp why we cant go in the middle and use middle DPI and get middle file size and good images. Is there any reason for this or are we missing some setting?

I will ask if they can test your example 🙂👍

**Reply by bsweeney** [reply]
I would think they could tweak the DPI of that image to better fit with the usage. They may have just bumped it up more than necessary to ensure good resolution. Getting the optimal size vs quality is an art and a skill.

**Reply by flamuren** [reply]
> I would think they could tweak the DPI of that image to better fit with the usage. They may have just bumped it up more than necessary to ensure good resolution. Getting the optimal size vs quality is an art and a skill.

Many thanks for this clarification. I have asked them this, lets hope they can do that 🙏
